vassal-app/src/main/java/VASSAL/build/module/map/Flare.java

```diff
@@ -0,0 +1,583 @@
+   */
+  @Override
+  public HelpFile getHelpFile() {
+    return HelpFile.getReferenceManualPage("Flare.htm");  //$NON-NLS-1$
+  }
+
+  /**
+   * Adds this component to a Buildable component. In this case, a Map.
+   * @param parent - the Map to add the Flare to.
+   */
+  public void addTo(final Buildable parent) {
+    if (parent instanceof Map) {
+      map = (Map) parent;
+      GameModule.getGameModule().addCommandEncoder(this);
+      map.addDrawComponent(this);
+      map.addLocalMouseListener(this);
+    }
+    else {
+      ErrorDialog.dataWarning(new BadDataReport("Flare - can only be added to a Map. ", reportFormat.getFormat()));
+    }
+  }
+
+  /**
+   * Removes this component from a Buildable parent.
+   * @param parent - the Map to remove the Flare from.
+   */
+  public void removeFrom(final Buildable parent) {
+    if (parent instanceof Map) {
+      GameModule.getGameModule().removeCommandEncoder(this);
+    }
+  }
+
+  private double os_scale = 1.0;
+
+  private volatile float animfrac;
+
+  private void repaintArea() {
+    map.repaint(new Rectangle(
+            (int)(clickPoint.x - circleSize / 2.0 - 2 * STROKE * os_scale),
+            (int)(clickPoint.y - circleSize / 2.0 - 2 * STROKE * os_scale),
+            (int)(circleSize + 4 * STROKE * os_scale + 0.5),
+            (int)(circleSize + 4 * STROKE * os_scale + 0.5)
+    ));
+  }
+
+  /**
+   * Animator to loop the Flare animation. Use the LOOP behavior so that it's always shrinking bullseye rings.
+   */
+  private final Animator animator = new Animator(0, 1, Animator.RepeatBehavior.LOOP, new TimingTargetAdapter() {
+    @Override
+    public void begin() {
+      active = true;
+      animfrac = 0.0f;
+      repaintArea();
+    }
+
+    @Override
+    public void timingEvent(float fraction) {
+      animfrac = fraction;
+      repaintArea();
+    }
+
+    @Override
+    public void end() {
+      active = false;
+      repaintArea();
+    }
+  });
+
+  /**
+   * Start the Flare animation.
+   * @param isLocal - true if this Flare was launched on this client (i.e. by this player). Otherwise, center on
+   *                  the ping location if user preferences are set for centering on opponent moves.
+   */
+  public void startAnimation(final boolean isLocal) {
+    if (map == null) {
+      return; // Means we already mentioned BadModuleData earlier. Now we simply avoid crashing.
+    }
+
+    if (!isLocal) {
+      if (GlobalOptions.getInstance().centerOnOpponentsMove()) {
+        map.centerAt(clickPoint);
+      }
+    }
+
+    animator.stop();
+    animate = pulses > 0 && pulsesPerSec > 0;
+    animator.setRepeatCount(Math.max(pulses, 1));
+    animator.setDuration(1000 / Math.max(pulsesPerSec, 1));
+    animator.start();
+  }
+
+  /**
+   * Draw the Flare at current animation frame
+   * @param g - Graphics
+   * @param map - Map component
+   */
+  public void draw(final Graphics g, final Map map) {
+    if (active && clickPoint != null) {
```

Comment:
I'd invert this condition and do an early return, then the whole method doesn't need to be indented an extra level:
```
if (!active || clickPoint == null) {
  return;
}

// rest of the code
```